From 13af81da96c4b706e3262f1424d1d26b3db315d1 Mon Sep 17 00:00:00 2001
From: Andrew Marshall <andrew@johnandrewmarshall.com>
Date: Wed, 6 Nov 2024 17:24:11 -0500
Subject: [PATCH 1/2] openvdb_11: init at 11.0.0

There are many breaking changes in v12, which some consumers are not
ready for and would require complex changes. This is expected to go away
once it becomes unused in nixpkgs.
---
 pkgs/development/libraries/openvdb/11.nix | 19 +++++++++++++++++++
 pkgs/top-level/all-packages.nix           |  1 +
 2 files changed, 20 insertions(+)
 create mode 100644 pkgs/development/libraries/openvdb/11.nix

diff --git a/pkgs/development/libraries/openvdb/11.nix b/pkgs/development/libraries/openvdb/11.nix
new file mode 100644
index 0000000000000..a65bb286ae2e2
--- /dev/null
+++ b/pkgs/development/libraries/openvdb/11.nix
@@ -0,0 +1,19 @@
+{
+  lib,
+  fetchFromGitHub,
+  openvdb,
+}:
+
+openvdb.overrideAttrs (old: rec {
+  name = "${old.pname}-${version}";
+  version = "11.0.0";
+  src = fetchFromGitHub {
+    owner = "AcademySoftwareFoundation";
+    repo = "openvdb";
+    rev = "v${version}";
+    sha256 = "sha256-wDDjX0nKZ4/DIbEX33PoxR43dJDj2NF3fm+Egug62GQ=";
+  };
+  meta = old.meta // {
+    license = lib.licenses.mpl20;
+  };
+})
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index 55968eed13554..428a42210da40 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -22214,6 +22214,7 @@ with pkgs;
   zunclient = with python311Packages; toPythonApplication python-zunclient;
 
   openvdb = callPackage ../development/libraries/openvdb { };
+  openvdb_11 = callPackage ../development/libraries/openvdb/11.nix { };
 
   openvr = callPackage ../by-name/op/openvr/package.nix {
     inherit (darwin.apple_sdk.frameworks) Foundation AppKit;

From 33e09c6eea89ae1d1145f2f4527f98fd7b87865b Mon Sep 17 00:00:00 2001
From: Andrew Marshall <andrew@johnandrewmarshall.com>
Date: Wed, 6 Nov 2024 17:25:29 -0500
Subject: [PATCH 2/2] blender: fix build by using openvdb_11

openvdb is now v12, and has many breaking API changes. Upstream has not
yet adapted to them yet, so there is no patch to backport. Further,
OpenVDB 12 is not currently part of the anticipated upstream library
updates for Blender 4.3 or 4.4.
---
 pkgs/applications/misc/blender/default.nix | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pkgs/applications/misc/blender/default.nix b/pkgs/applications/misc/blender/default.nix
index 7d89339da1026..e2af0b3f93384 100644
--- a/pkgs/applications/misc/blender/default.nix
+++ b/pkgs/applications/misc/blender/default.nix
@@ -64,7 +64,7 @@
   openjpeg,
   openpgl,
   opensubdiv,
-  openvdb,
+  openvdb_11,
   openxr-loader,
   pkg-config,
   potrace,
@@ -276,7 +276,7 @@ stdenv.mkDerivation (finalAttrs: {
       openjpeg
       openpgl
       (opensubdiv.override { inherit cudaSupport; })
-      openvdb
+      openvdb_11
       potrace
       pugixml
       python3
